OneCommunityGlobal/HighestGoodNetworkApp
# Kaikanes LessonForm Tags Functionality

# Description
Added Tags functionality to the lessonform. 
Display all projects and roles in dropdown
Added Lessonform button to the project dashboard.

## Related PRS (if any):
building off of theses previous PR's
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1477 (figma design)
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1577 (drag and drop)
…

## Main changes explained:
Fetched projects and roles so the dropdowns displays all the  projects/roles in the database
Users can now enter tags by typing and pressing enter. 
When a user selects a project from the dropdown it automatically adds that project as a tag.
Added a lesson button on the bmdashboard that displays a project specific form that has the belonging project locked.
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as anyone
4. go to http://localhost:3000/bmdashboard/  -> select a project and click the new lessonform button
5. verify the project specific form displays the correct projectID. (just using projectID dummy data for now until /bmdashboard is displaying real data.
6. test the new tags functionality. tr to break it. users should not be able to add duplicate tags. 
7. go to http://localhost:3000/bmdashboard/lessonform/ and test the non project specific form also


## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114305309/4fd8e3a6-e1a7-4b84-add7-e8d1f748cdb4





## Note:
When selecting a specific project from the dropdown the displayed title is different because we are using the projectId not the project name.  Later will probably do a fetch or pass more props.
<img width="505" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114305309/ec2395cf-9266-4044-8540-c7da8ffb971d">

<img width="444" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114305309/0108b4de-1b17-4f58-b7c6-bd583ab0e8ba">
